src/psyclone/psyir/nodes/node.py

```diff
@@ -1077,21 +1077,27 @@ def _insert_schedule(self, children=None, ast=None):
             sched.ast = ast
         return sched
 
-    def find_symbol_table(self):
-        '''
-        :returns: the symbol table attached to the nearest ancestor \
-            node (including self).
-        :rtype: :py:class:`psyclone.psyir.symbols.SymbolTable`
+    @property
+    def scope(self):
+        '''Schedule and Container nodes allow symbols to be scoped via an
+        attached symbol table. This property returns the closest
+        ancestor Schedule or Container node including self.
+
+        :returns: the closest ancestor Schedule or Container node.
+        :rtype: :py:class:`psyclone.psyir.node.Node`
 
-        :raises InternalError: if no symbol table is found.
+        :raises InternalError: if there is no Schedule or Container \
+            ancestor.
 
         '''
+        from psyclone.psyir.nodes import Schedule, Container
         current = self
-        while current and not hasattr(current, "symbol_table"):
+        while current and not isinstance(current, (Container, Schedule)):
             current = current.parent
         if current:
-            return current.symbol_table
-        raise InternalError("Symbol table not found in any ancestor nodes.")
+            return current
+        raise InternalError(
+            "None of this node's ancestors are Container or Schedule nodes.")
```

Comment:
I imagine scope will usually be used together with other properties/methods in the same line. So maybe it will be helpful to me more explicit that is the scope property that failed? Something like `"Could't find the scope of '{0}' as none of its ancestors are Container or Schedule nodes.".format(self)`

Reply:
Yes, that is nicer, thanks.